artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -825,8 +826,14 @@ protected ReadableBuffer createDeliveryCopy(int deliveryCount, DeliveryAnnotatio
       }
 
       writeDeliveryAnnotationsForSendBuffer(result, deliveryAnnotations);
-      // skip existing delivery annotations of the original message
-      duplicate.position(encodedHeaderSize + encodedDeliveryAnnotationsSize);
+
```

Comment:
I'm not convinced you need to change this after having adjusted the encodedHeraderSize value to account for it not being at the start of the buffer.  I must say that it feels a bit odd to be allowing wrongly encoded AMQP message format zero messages through in the first place, we probably should have been catching that and rejecting the messages really.

Reply:
I created a couple simple tests for this and just the one line fix [here](https://github.com/tabish121/activemq-artemis/commit/7f19cfc965d06c08217967fcf5bfb6d28d5058da) and that seems to be all that's needed:

Reply:
@tabish121 this is working around wrongly encoded messages from the older version.

in case an user has messages wrongly encoded, they could just switch to the newer version that would know how to deal with it.

What is the alternative?

Reply:
and this is the reason I expanded the compatibility-tests.. if you generated the messages in 2.33, stopped both brokers with accumulated messages, the clients won't know how to parse them because they are wrongly encoded.

Reply:
The compatibly tests are fairly incomprehensible so I can't really tell what it is doing or if it is or isn't reproducing anything.  I added a link to commit that tests the basic handling of message encoded with delivery annotations ahead of headers and showed that the one line change to fix the encoded header size sorts that scenario.

Reply:
@tabish121 the compatibility tests is creating the situation with 2.33.. and consuming the message in HEADER/main


basically creating the real condition where it happened.

Reply:
and the messages are not consumed properly with your fix.. I just tested it.

Reply:
Then it would be good to create a test case in the AMQPMessageTest that is reproducing the formatting and verifying a fix, you fix doesn't look correct to me but without knowing what you are actually reproducing its hard to judge

Reply:
The test is creating the wrong encoding in 2.33 and consuming it in main using real bits from 2.33, just like the customer case.

I will work on the AMQPMessageTest.

Reply:
@tabish121 my fix is only applied when the header is switched with the delivery annotation, an aberration that shouldn't actually ever happen, but it did.

Reply:
@tabish121 the test you wrote is validating if the parsing would work correctly.. however with your fix the client is not being to actually receive the message.

You would need to write a test with the wrong encoding and have an actual client receiving the message.

Reply:
There should be no need for a client to test this, just the test decodes the message based on what you said is the issue, without knowing more I can't write a better test but that doesn't mean it isn't testable without a client.